File: no 2/php/hardware.php
<?php

class Hardware extends Product{
	private $brand;
	private $model;

	// set dan get 

	function __construct(){
	}

	function setBrand($brand){
		$this->brand = $brand;
	}
	function getBrand(){
		return $this->brand;
	}

	function setModel($model){
		$this->model = $model;
	}
	function getModel(){
		return $this->model;
	}

	function __destruct(){

	}

}

?>

File: no 2/cpp/hardware.cpp

class Hardware: public Memory, public Product{

	private:
		string brand;
		string model;

	public:
		Hardware(){
		}

		// set dan get dari private
		void setBrand(string brand){
			this->brand = brand;
		}
		string getBrand(){
			return this->brand;
		}

		void setModel(string model){
			this->model = model;
		}
		string getModel(){
			return this->model;
		}


		~Hardware(){
		}

};